crates/bevy_pbr/src/environment_map/filter.wgsl

```diff
@@ -0,0 +1,187 @@
+@group(0) @binding(6) var text_out5: texture_storage_2d_array<rg11b10float, write>;
+@group(0) @binding(7) var text_out6: texture_storage_2d_array<rg11b10float, write>;
+#else
+@group(0) @binding(1) var text_out0: texture_storage_2d_array<rgba16float, write>;
+@group(0) @binding(2) var text_out1: texture_storage_2d_array<rgba16float, write>;
+@group(0) @binding(3) var text_out2: texture_storage_2d_array<rgba16float, write>;
+@group(0) @binding(4) var text_out3: texture_storage_2d_array<rgba16float, write>;
+@group(0) @binding(5) var text_out4: texture_storage_2d_array<rgba16float, write>;
+@group(0) @binding(6) var text_out5: texture_storage_2d_array<rgba16float, write>;
+@group(0) @binding(7) var text_out6: texture_storage_2d_array<rgba16float, write>;
+#endif
+@group(0) @binding(8) var trilinear: sampler;
+@group(0) @binding(9) var<uniform> coeffs: array<array<array<array<vec4<f32>, 24>, 3>, 5>, 7>;
+
+fn get_dir(u: f32, v: f32, face: u32) -> vec3<f32> {
+    switch face {
+        case 0u: { return vec3(1.0, v, -u); }
+        case 1u: { return vec3(-1.0, v, u); }
+        case 2u: { return vec3(u, 1.0, -v); }
+        case 3u: { return vec3(u, -1.0, v); }
+        case 4u: { return vec3(u, v, 1.0); }
+        default { return vec3(-u, v, -1.0); }
+    }
+}
+
+@compute
+@workgroup_size(64, 1, 1)
+fn main(@builtin(global_invocation_id) global_id: vec3<u32>) {
+    var id = global_id;
+    var level = 0u;
+    if id.x < 128u * 128u {
+        level = 0u;
+    } else if id.x < 128u * 128u + 64u * 64u {
+        level = 1u;
+        id.x -= 128u * 128u;
+    } else if id.x < 128u * 128u + 64u * 64u + 32u * 32u {
+        level = 2u;
+        id.x -= 128u * 128u + 64u * 64u;
+    } else if id.x < 128u * 128u + 64u * 64u + 32u * 32u + 16u * 16u {
+        level = 3u;
+        id.x -= 128u * 128u + 64u * 64u + 32u * 32u;
+    } else if id.x < 128u * 128u + 64u * 64u + 32u * 32u + 16u * 16u + 8u * 8u {
+        level = 4u;
+        id.x -= 128u * 128u + 64u * 64u + 32u * 32u + 16u * 16u;
+    } else if id.x < 128u * 128u + 64u * 64u + 32u * 32u + 16u * 16u + 8u * 8u + 4u * 4u {
+        level = 5u;
+        id.x -= 128u * 128u + 64u * 64u + 32u * 32u + 16u * 16u + 8u * 8u;
+    } else if id.x < 128u * 128u + 64u * 64u + 32u * 32u + 16u * 16u + 8u * 8u + 4u * 4u + 2u * 2u {
+        level = 6u;
+        id.x -= 128u * 128u + 64u * 64u + 32u * 32u + 16u * 16u + 8u * 8u + 4u * 4u;
+    } else {
+        return;
+    }
+
+    id.z = id.y;
+    let res = 128u >> level;
+    id.y = id.x / res;
+    id.x -= id.y * res;
+
+    let u = (f32(id.x) * 2.0 + 1.0) / f32(res) - 1.0;
+    let v = -(f32(id.y) * 2.0 + 1.0) / f32(res) + 1.0;
+
+    let dir = get_dir(u, v, id.z);
+    let frame_z = normalize(dir);
+    let adir = abs(dir);
+
+    var color = vec4(0.0);
+    for (var axis = 0u; axis < 3u; axis++) {
+        let other_axis0 = 1u - (axis & 1u) - (axis >> 1u);
+        let other_axis1 = 2u - (axis >> 1u);
```

Comment:
I feel like if you're switching over `axis` below anyway you might as well just set `other_axis0` and `other_axis1` explicitly in a switch statement instead of using these bit tricks.